src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -780,13 +819,29 @@ export class SearchView extends ViewPane {
 		e.browserEvent.preventDefault();
 		e.browserEvent.stopPropagation();
 
+		const match = e.element;
+		let locations: Location[] = [];
+		if (match instanceof FileMatch) {
+			locations = match.matches().map(toLocation);
+		} else if (match instanceof Match) {
+			locations = [toLocation(match)];
+		} else if (match instanceof FolderMatch) {
+			locations = flatten(match.matches().map(fileMatch => fileMatch.matches().map(toLocation)));
+		}
+
+		let context = {
```

Comment:
@jrieken are you good with this? Making the context which could be a common type `{ locations: [...] }`, or we could name it `location` but give it an array. Or we could use `{ location: Location }[]`, but we have to think about multiselect in the future.

Reply:
Also note that `renderableMatch` is like an "internal context" that the argument processor removes.

Reply:
For multiselect the custom tree does the following: the first element is always the active tree item, the second element is all selected items. So, you select 7 elements and right click on the 3rd, then it is `[3rd element, [7 elements]]`

For the actual data type I am unsure. A location seems to be working well but not differentiating between a single match and group of matches (file) and a group of groups (folder) will likely not make everyone happy.

Reply:
Oh ok, I didn't know that. I think the explorer is just always a list of URIs.

Someone could reverse engineer the groupings from our flat list if they wanted to. I still think it's a workable solution but yeah it's not what I would design from scratch just for search. Besides this simple common type, we could reuse `TextSearchMatch` from the findTextInFiles proposed api. I'd rather use something that matches what's in the UI more directly though. And that won't be stabilized soon.